filesource-common/src/main/java/io/cdap/plugin/common/FileSourceConfig.java

```diff
@@ -131,11 +136,16 @@ public FileSourceConfig(String referenceName, @Nullable String fileRegex, @Nulla
     this.schema = schema;
   }
 
-  protected void validate() {
-    getFileSystemProperties();
+  protected void validate(FailureCollector collector) {
+    try {
+      getFileSystemProperties();
+    } catch (IllegalArgumentException e) {
+      collector.addFailure(e.getMessage(), null).withConfigProperty(FILE_SYSTEM_PROPERTIES);
+    }
     if (!CombinePathTrackingInputFormat.class.getName().equals(inputFormatClass) && pathField != null) {
-      throw new IllegalArgumentException("pathField can only be used if inputFormatClass is " +
-                                           CombinePathTrackingInputFormat.class.getName());
+      collector.addFailure("Path field can only be used if inputFormatClass is " +
```

Comment:
Curious if the `CombinePathTrackingInputFormat.class.getName()` adds a period at the end. Let me know if you have manually verified it.